File: Template Method/Ruby/TwoFactorEmail.rb
require_relative "TwoFactorAuthentication"

class TwoFactorEmail < TwoFactorAuthentication
    def getData(username)
        puts "Getting email data for #{username}"
    end

    def sendNotification(username)
        puts "Sending email notification to #{username}"
    end

    def match(username)
        puts "Matching #{username} with the code"
    end
end

File: Template Method/Javascript/TwoFactorEmail.js
import TwoFactorAuthentication from "./TwoFactorAuthentication.js";

class TwoFactorEmail extends TwoFactorAuthentication {
  getData(username) {
    console.log(`Getting email data for ${username}`);
  }

  sendNotification(username) {
    console.log(`Sending email notification to ${username}`);
  }

  match(username) {
    console.log(`Matching ${username} email code`);
  }
}

export default TwoFactorEmail;